modules/tinymce/src/core/test/ts/browser/dom/BundledCssTest.ts

```diff
@@ -0,0 +1,256 @@
+  const skinUiJs = (skin: 'oxide' | 'oxide-dark', shadowDom: boolean) => skinUiDir(skin) + `skin${shadowDom ? '.shadowdom' : ''}.js`;
+
+  const skinUiContentCss = (skin: 'oxide' | 'oxide-dark', inline: boolean, withSuffix: boolean) => skinUiDir(skin) + `content${inline ? '.inline' : ''}${withSuffix ? '.min' : ''}.css`;
+  const skinUiContentJs = (skin: 'oxide' | 'oxide-dark', inline: boolean) => skinUiDir(skin) + `content${inline ? '.inline' : ''}.js`;
+
+  const styleTagsToKeys = (styleTags: SugarElement<HTMLStyleElement>[]) => {
+    return Arr.map(styleTags, (tag) => tag.dom.dataset.mceKey);
+  };
+
+  const linkTagsToKeys = (linkTags: SugarElement<HTMLLinkElement>[], basePath: string) => {
+    return Arr.map(linkTags, (tag) => tag.dom.href.replace(basePath, ''));
+  };
+
+  const getCssTags = (container: SugarElement<HTMLHeadElement | ShadowRoot>) => {
+    const style =
+      SelectorFilter.descendants<HTMLStyleElement>(
+        container,
+        'style'
+      );
+    const link =
+      SelectorFilter.descendants<HTMLLinkElement>(
+        container,
+        'link'
+      );
+
+    return {
+      style,
+      link
+    };
+  };
+
+  const getIframeCSSTags = (editor: Editor) =>
+    getCssTags(SugarShadowDom.getStyleContainer(SugarShadowDom.getRootNode(TinyDom.body(editor))));
+
+  const getCSSContainerTags = (editor: Editor) =>
+    getCssTags(SugarShadowDom.getStyleContainer(SugarShadowDom.getRootNode(TinyDom.targetElement(editor))));
+
+  const pCreateEditor = async (shadowDom: boolean, settings: Record<string, any>) => {
+    if (shadowDom) {
+      const shadowHost = SugarElement.fromTag('div', document);
+      Insert.append(SugarBody.body(), shadowHost);
+      const sr = SugarElement.fromDom(shadowHost.dom.attachShadow({ mode: 'open' }));
+      const editorDiv = SugarElement.fromTag('div', document);
+      Insert.append(sr, editorDiv);
+
+      const cleanup = () => {
+        Remove.remove(shadowHost);
+      };
+      const editor = await McEditor.pFromElement<Editor>(editorDiv, {
+        ...settings
+      });
+      return [ editor, cleanup ] as const;
+    } else {
+      const editor = await McEditor.pFromSettings<Editor>({
+        ...settings
+      });
+      const cleanup = Fun.noop;
+      return [ editor, cleanup ] as const;
+    }
+  };
+
+  for (const inline of [ false, true ]) {
+    context('inline: ' + inline, () => {
+      for (const shadowdom of [ false, true ]) {
+        context('shadowdom: ' + shadowdom, () => {
+          for (const contentCss of [ 'default', 'dark' ] as const) {
+            context('contentCSS: ' + contentCss, () => {
+              for (const skin of [ 'oxide', 'oxide-dark' ] as const) {
+                context('skin: ' + skin, () => {
+                  const oxideUiSkinMap: Record<string, string> = {
```

Comment:
Hey Luke, is it important that we define `oxideUiSkinMap` inside of the nested loops like this?
If we define it outside then we can just create the object once and reference it. Unless I'm mistaken, we should have it in the scope and it doesn't look like we are modifying it.

Reply:
If we were outside a test file, I would be more worried about this. Although, for test files, I think visual readability is more important than absolute importance.
In any case, I can make the change 👍